lib/endpointer/cacher.rb

```diff
@@ -32,14 +34,18 @@ def create_cache_container(resource, response)
       Endpointer::CacheContainer.new(resource, response, Time.now.utc)
     end
 
-    def retrieve_cache_container(resource)
+    def retrieve_cache_container(cache_key)
       begin
-        YAML.load(File.read(File.join(@path, "#{resource.id}.yml")))
+        YAML.load(File.read(File.join(@path, cache_key)))
       rescue
         raise Endpointer::Errors::CachedItemNotFoundError
       end
```

Comment:
that's a catch all exception. You might have a lot of reason why the yaml file can't be loaded, but by catching any exception that might lead to unexpected behaviour.